streams/src/main/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImpl.java

```diff
@@ -136,13 +137,16 @@ public SessionWindowedCogroupedKStream<K, VOut> windowedBy(final SessionWindows
     private KTable<K, VOut> doAggregate(final Initializer<VOut> initializer,
                                         final NamedInternal named,
                                         final MaterializedInternal<K, VOut, KeyValueStore<Bytes, byte[]>> materializedInternal) {
+        final boolean isOutputVersioned = materializedInternal != null
+            && materializedInternal.storeSupplier() instanceof VersionedBytesStoreSupplier;
         return aggregateBuilder.build(
             groupPatterns,
             initializer,
             named,
             new KeyValueStoreMaterializer<>(materializedInternal).materialize(),
             materializedInternal.keySerde(),
             materializedInternal.valueSerde(),
-            materializedInternal.queryableStoreName());
+            materializedInternal.queryableStoreName(),
+            isOutputVersioned);
```

Comment:
Think we can just pass in `materializedInternal.storeSupplier() instanceof VersionedBytesStoreSupplier` directly